tools/unit-test/slang-unit-test.h

```diff
@@ -58,18 +58,24 @@ class UnitTestRegisterHelper
     UnitTestRegisterHelper(const char* name, UnitTestFunc testFunc);
 };
 
+class AbortTestException {};
+
 typedef IUnitTestModule* (*UnitTestGetModuleFunc)();
 
 #define SLANG_UNIT_TEST(name) \
-void name(UnitTestContext* unitTestContext); \
+void _##name##_impl(UnitTestContext* unitTestContext); \
+void name(UnitTestContext* unitTestContext)\
+{\
+    try { _##name##_impl(unitTestContext); } catch (AbortTestException){} \
```

Comment:
Not super important here, but `catch (AbortTestException&)`  would arguably be preferred.